src/core/builder/hardhat.ts

```diff
@@ -16,52 +16,36 @@ export class HardhatBuilder implements IBuilder {
   }): Promise<{
     contracts: ContractPayload[];
   }> {
-    let hardhatPath: string | undefined;
-    try {
-      hardhatPath = require.resolve("hardhat");
-    } catch (e) {
-      logger.error(
-        "failed to load hardhat runtime, please install hardhat: npm i -g hardhat"
-      );
-      process.exit(1);
+    if (options.clean) {
+      logger.info("Running hardhat clean");
+      execSync("npx hardhat clean");
     }
-    logger.debug("Hardhat path found", hardhatPath);
-    let hre: HardhatRuntimeEnvironment | undefined;
-    try {
-      hre = require(hardhatPath);
-      logger.debug("userconfig paths", hre?.userConfig.paths);
 
-      if (options.clean) {
-        logger.info("Cleaning before compiling");
-        await hre?.run("clean");
-      }
+    logger.info("Compiling...");
+    execSync("npx hardhat compile");
```

Comment:
how does this work with custom paths? it should fail at this point no? sinnce its not loading the user conf before compiling

Reply:
nah, each npx hardhat command is stand alone here. they just "happen" to all run in the same context, we let hardhat do all its own thing, we just "trick" it to log out the config it happens to be running with in the other task

Reply:
aaah i see of course, npx hardhat still manages to read its conf, and then you grab it from global env